core/server/apps/amp/lib/helpers/amp_components.js

```diff
@@ -0,0 +1,29 @@
+// # Amp Components Helper
+// Usage: `{{amp_components}}`
+//
+// Reads through the AMP HTML and adds neccessary scripts
+// for each extended component
+// If more components need to be added, we can add more scripts.
+// Here's the list of all supported extended components: https://www.ampproject.org/docs/reference/extended.html
+var hbs             = require('express-hbs');
+
+function ampComponents() {
+    var components = [],
+        html = this.post.html || this.html;
+
+    if (html.indexOf('.gif') > 0) {
+        components.push('<script async custom-element="amp-anim" src="https://cdn.ampproject.org/v0/amp-anim-0.1.js"></script>');
+    }
+
+    if (html.indexOf('iframe') > 0) {
+        components.push('<script async custom-element="amp-iframe" src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"></script>');
+    }
+
+    if (html.indexOf('form') > 0 || html.indexOf('input') > 0) {
+        components.push('<script async custom-element="amp-form" src="https://cdn.ampproject.org/v0/amp-form-0.1.js"></script>');
+    }
+
+    return new hbs.handlebars.SafeString(components.join('\n    ').trim());
```

Comment:
I _borrowed_ it from [{{ghost_head}}](https://github.com/TryGhost/Ghost/blob/master/core/server/helpers/ghost_head.js#L131), but it works fine without the whitespace (initially, I was assuming it is needed for the indentation).